### PR TITLE
Store Orders: Add fees to an existing order

### PR DESCRIPTION
**Pull request by ryelle**
This PR adds a flow to add fees to an existing order. There are now two buttons under the items table, for adding fees and products. This PR only adds the  `+ Fee` dialog.

order-with-buttons

Once clicked, + Fee opens this modal (the misaligned text input will be fixed with #19130). The Add Fee button is disabled until content is entered in both the name and value inputs. The mockup initially asked for just a value, with the option of flat fee or percentage fee — we need a name as well, so I've added the input for that, and percentage fees aren't supported by the API, so I'm just asking for a flat fee.

fee-modal

Once added, the new fee shows up in the orders table, with the ability to delete it if necessary:

added-fee

**To test**

- View your orders, and click into a single order. Toggle on the edit state by clicking Edit Order
- Click the `+ Fee` button, make sure the fee modal opens
- Note that save is disabled until you fill out the form, but you can close the modal by clicking Cancel
- Enter a few different values for fees, make sure they're added to the order as you'd expect

Note: This doesn't save yet, if you go to save the order you'll get an error, `woocommerce_rest_invalid_item_id`– "Order item ID provided is not associated with order." This is due to the fee's placeholder ID, and is fixed by using the removeTemporaryIds function that's added in #19128

### CODE REVIEW

File: client/extensions/woocommerce/app/order/order-details/fee-dialog.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			} );
+		}
+		this.props.closeDialog();
+	};
+
+	render() {
+		const { closeDialog, isVisible, order, translate } = this.props;
+		const dialogClass = 'woocommerce order-details__dialog'; // eslint/css specificity hack
+
+		const canSave = this.state.name && this.state.total;
```
Comment (allendav) [inline client/extensions/woocommerce/app/order/order-details/fee-dialog.js]: Please trim the name before testing for save-ability

**Comment by allendav** [comment]
Do we have to say Fee name and Fee value, or could we just say Name and Value since the context for the user should be clear for such a simple dialog?

File: client/extensions/woocommerce/app/order/order-details/add-items.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	toggleDialog = type => () => {
+		this.setState( { showDialog: type } );
+	};
+
+	render() {
+		const { translate } = this.props;
+		return (
+			<div className="order-details__actions">
+				<Button borderless onClick={ this.toggleDialog( 'product' ) }>
+					{ translate( '+ Add Product' ) }
```
Comment (allendav) [inline client/extensions/woocommerce/app/order/order-details/add-items.js]: Optional: What would you think about adding the + using CSS instead, both here and below?

File: client/extensions/woocommerce/app/order/order-details/fee-dialog.js (hunk 1 above)
Comment (allendav) [inline client/extensions/woocommerce/app/order/order-details/fee-dialog.js]: Nit: You can enter a name (e.g. 'Blerg') and then a fractional cent fee (e.g. 0.001) and the Add Fee button will light up (but doesn't actually add the fee.) Maybe tie the button state to both name and a minimum fee of 0.01?

**Comment by kellychoffman** [comment]
Instead of an inline text `+` can we use the `add-outline` gridicon?

File: client/extensions/woocommerce/app/order/order-details/add-items.js (hunk 2 above)
Comment (timmyc) [reply client/extensions/woocommerce/app/order/order-details/add-items.js]: +1 on @kellychoffman's idea to use gridicons here.

**Comment by ryelle** [comment]
I've added the gridicon in place of `+`, and addressed the other issues around the fee name/value & saving. The right-alignment of text was fixed in #19130, and I'm going to leave mobile styling alone since we talked about doing a mobile sprint on orders later (https://github.com/Automattic/wp-calypso/pull/19047#issuecomment-338759884).

**Comment by ryelle** [comment]
Also, now that #19128 is merged, saving your added fees should work 🎉

**Pull request merged**